### PR TITLE
Migrate to configuration management

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/17386

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Review by rossf7 — APPROVED** [review]
LGTM

File: helm/chart-operator/values.yaml [hunk 1]
```diff
@@ -70,3 +73,5 @@ tiller:
 
 verticalPodAutoscaler:
   enabled: true
+
+isManagementCluster: false
```
Comment (kopiczko) [inline helm/chart-operator/values.yaml:77]: Sneaky. We can do the same for all apps that rely on the `-unique` suffix.

File: helm/chart-operator/templates/deployment.yaml [hunk 2]
```diff
@@ -70,7 +70,11 @@ spec:
           value: "5"
       initContainers:
       - name: wait-for-quay
+        {{ if .Values.isManagementCluster }}
+        image: "{{ .Values.registry.domain }}/{{ .Values.image.name }}:{{ .Values.image.tag }}"
+        {{ else }}
         image: "{{ .Values.image.registry }}/{{ .Values.image.name }}:{{ .Values.image.tag }}"
+        {{ end }}
```
Comment (kopiczko) [inline helm/chart-operator/templates/deployment.yaml:77]: For my interest: why different images on MC and WC?

**Review by kopiczko — APPROVED** [review]

File: helm/chart-operator/templates/deployment.yaml (hunk 2 above)
Comment (corest) [reply helm/chart-operator/templates/deployment.yaml:77]: I don't know, but also don't want to investigate this now. Can be easily changed via config repo later

Comment (rossf7) [reply helm/chart-operator/templates/deployment.yaml:77]: We use `image.registry` in the default catalog. So I'd prefer to use that and remove some of the conditionals.

But I'm fine if we do that in batman once this migration is done.

https://github.com/giantswarm/installations/blob/80e965ba6c529a52f491932fb0972e851c29b374/giraffe/appcatalog/default-appcatalog-values.yaml#L10-L11

**Pull request merged**